### DISCUSSION TITLE
Pulp takes too long to solve linear programming problems

### DISCUSSION THREAD

**Discussion by zhanhang123** [coin-or/pulp#546] [Q&A]
Recently, I am trying to solve a linear programming problem. However, in some cases, this linear programming problem becomes a constrained programming problem. Its optimization objective is applicable to any result, that is, the result of the optimization function of any result is the best. However, at this time, the speed of solving becomes extremely slow. No service guarantees to get all the solutions, but the speed of getting some solutions is also worrying. What is the reason for this problem, and how can I deal with it to improve its speed

I asked a related question on stackoverflow [[pulp-solves-multiple-combinatorial-problems]](https://stackoverflow.com/q/72305412/12332375)，but so far there is no answer.

All right, what's the best way to do this. Thanks.

**Comment by MaxAVF** [comment]
I'm having a similar problem, in my case seems like the adding contraints is taking A LOT of time, talking about hours, and the solving part just take a few minutes. I have seen other people asking about this problem but didnt find a solution yet

**Reply by jerrylshaw** [reply]
It seems your best option is to contact the PuLP publisher to get a full refund of your purchase price.

**Reply by MaxAVF** [reply]
@jerrylshaw Cool, a new user with zero contributions to anything trying to be funny thinking this is reddit or something...
I was just trying to report a problem

**Comment by stumitchell** [comment]
did you try the suggestions here?
http://www.stuartmitchell.com/journal/2012/2/3/my-top-n-tips-for-python-coding-in-optimisation-1.html
